### PR TITLE
Cache busting

### PR DESCRIPTION
**Pull request by mjoyce91**
Add functionality to include randomly generated query parameters and cookies. Based on https://github.com/erkexzcx/stoppropaganda/issues/129

This the first meaningful thing I've written in Go so apologies for any ugly code 😸

### CODE REVIEW

**Comment by erkexzcx** [comment]
Great! Just so you know, there is bool data type too. I would specify like this:
```go
flagRandomQuery = fs.Bool("randomquery", false, "append random query to URLs to disrupt caching mechanisms")
```

And then usage:
```go
if *flagRandomQuery {
    ...
}
```

File: internal/stoppropaganda/websites.go [hunk 1]
```diff
@@ -275,6 +288,27 @@ func doSingleRequest(ws *Website, req *fasthttp.Request, resp *fasthttp.Response
 
 	resp.ShouldDiscardBody = true
 
+	if *flagUseQueryParams == 1 {
+		// clean up query string and cookies from previous request
+		ws.req.URI().SetQueryString("")
```
Comment (erkexzcx) [inline internal/stoppropaganda/websites.go]: Check function arguments - there is `req` passed already, which is copy of `ws.req`. Also `ws.req`
 is not used at all in this function.

A bit off topic - accessing `ws.req` like this is not thread-safe and leads to race conditions (app will crash). This is the reason why there is function argument `req` which is copy of `ws.req` which was copied in a thread-safe manner before calling this function.

Comment (mjoyce91) [reply internal/stoppropaganda/websites.go]: Updated

File: internal/stoppropaganda/websites.go [hunk 2]
```diff
@@ -275,6 +288,27 @@ func doSingleRequest(ws *Website, req *fasthttp.Request, resp *fasthttp.Response
 
 	resp.ShouldDiscardBody = true
 
+	if *flagRandomQuery {
+		// clean up query string and cookies from previous request
+		req.URI().SetQueryString("")
```
Comment (erkexzcx) [inline internal/stoppropaganda/websites.go]: You don't need to clean anything here. :)

Copy of request pointer (*fasthttp.Request) is being done here: https://github.com/erkexzcx/stoppropaganda/blob/main/internal/stoppropaganda/websites.go#L239

You will always have a fresh, original and unmodified copy of request in each function call, so you can remove those 2 lines.

Comment (mjoyce91) [reply internal/stoppropaganda/websites.go]: Ah okay! This was needed when I was modifying `ws` directly but not any more

**Comment by erkexzcx** [comment]
I've fixed the following for you:
1. Reuse `fastrand`. This removes `rand` and `math/rand` package dependencies.
2. Do not randomise the count of query keys/cookies to save resources and achieve the same effect. It will always be 1.

This is how URLs now look like :)
```
2022/03/13 10:53:57 https://mvd.gov.ru/?dofrwwbygokcsye=vftkhfpkkgly
2022/03/13 10:53:57 https://mvd.gov.ru/?zzcvgnwzvoeuc=gzluhfly
2022/03/13 10:53:57 https://mvd.gov.ru/?yyhwrpk=hkobhjnmmz
2022/03/13 10:53:57 https://mvd.gov.ru/?eagskyw=oychpbcskqecinjhpzd
2022/03/13 10:53:57 https://mvd.gov.ru/?sfzukxyaely=ffewbfq
2022/03/13 10:53:57 https://api.pikabu.ru/?zljcssleuobp=zyskqui
2022/03/13 10:53:57 https://api.pikabu.ru/?htfulbfg=vccuxkwsjwbtlrr
2022/03/13 10:53:57 https://api.pikabu.ru/?pqtluhfeaoqe=bktwchngpnmnfxli
2022/03/13 10:53:57 https://api.pikabu.ru/?nkakeregmpkkaqji=vzyejh
```

**Pull request merged**